crates/goose/src/providers/openai.rs

```diff
@@ -160,6 +162,31 @@ impl OpenAiProvider {
             .await?;
         handle_response_openai_compat(response).await
     }
+
+    // Core completion logic that takes a model config
+    async fn complete_with_model(
+        &self,
+        model_config: &ModelConfig,
+        system: &str,
+        messages: &[Message],
+        tools: &[Tool],
+    ) -> Result<(Message, ProviderUsage), ProviderError> {
+        let payload = create_request(model_config, system, messages, tools, &ImageFormat::OpenAi)?;
+
+        let json_response = self.post(&payload).await?;
+
+        let message = response_to_message(&json_response)?;
+        let usage = json_response
+            .get("usage")
+            .map(get_usage)
+            .unwrap_or_else(|| {
+                tracing::debug!("Failed to get usage data");
+                Usage::default()
+            });
+        let model = get_model(&json_response);
+        emit_debug_trace(model_config, &payload, &json_response, &usage);
+        Ok((message, ProviderUsage::new(model, usage)))
+    }
```

Comment:
this was previously existing, but now that we pass in the model, no need to fish it out of the json response